src/main/java/org/opentripplanner/netex/mapping/FlexStopsMapper.java

```diff
@@ -58,43 +58,57 @@ class FlexStopsMapper {
    * dependent on a key/value in the NeTEx file, until proper NeTEx support is added.
    */
   StopLocation map(FlexibleStopPlace flexibleStopPlace) {
-    Object area = flexibleStopPlace
-      .getAreas()
-      .getFlexibleAreaOrFlexibleAreaRefOrHailAndRideArea()
-      .get(0);
-    if (!(area instanceof FlexibleArea)) {
-      issueStore.add(
-        Issue.issue(
-          "UnsupportedFlexibleStopPlaceAreaType",
-          "FlexibleStopPlace %s contains an unsupported area.",
+    var areas = flexibleStopPlace.getAreas().getFlexibleAreaOrFlexibleAreaRefOrHailAndRideArea();
+
+    List<StopLocation> stops = new ArrayList<>();
+
+    for (var area : areas) {
+      if (!(area instanceof FlexibleArea flexibleArea)) {
+        issueStore.add(
+          Issue.issue(
+            "UnsupportedFlexibleStopPlaceAreaType",
+            "FlexibleStopPlace %s contains an unsupported area.",
+            flexibleStopPlace.getId()
+          )
+        );
+        LOG.warn(
+          "FlexibleStopPlace {} not mapped. Hail and ride areas are not currently supported.",
           flexibleStopPlace.getId()
-        )
-      );
-      LOG.warn(
-        "FlexibleStopPlace {} not mapped. Hail and ride areas are not currently supported.",
-        flexibleStopPlace.getId()
-      );
-      return null;
-    }
+        );
+        return null;
+      }
 
-    Optional<KeyValueStructure> flexibleAreaType = Optional.empty();
-    if (flexibleStopPlace.getKeyList() != null) {
-      flexibleAreaType =
-        flexibleStopPlace
-          .getKeyList()
-          .getKeyValue()
-          .stream()
-          .filter(k -> k.getKey().equals(FLEXIBLE_STOP_AREA_TYPE_KEY))
-          .findFirst();
+      String flexibleAreaType = null;
```

Comment:
The following code in the loop doesn't use `area` , only `flexibleStopPlace` directly. Effectively the method always returns on the first iteration, leading to not all areas being type checked above.

Suggestion is to move the type check to a separate method or at least move out the following code from the loop.

Reply:
This is a question. I was under the impression that the key should be in the FlexibleArea, not the FlexibleStopPlace, as that makes it impossible to combine different types of areas inside a stop place, which is a goal of ours.

Reply:
I think the type key/value should be possible to specify on both FlexibleStopPlace and FlexibleArea, but if there it is on FlexibleArea, then that should have priority. Then we have backwards compatibility with existing implementations.